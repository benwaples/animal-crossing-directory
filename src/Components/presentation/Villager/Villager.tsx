import React from 'react'
import { Link } from 'react-router-dom'
// import styles from './Villager.css'

type Props = {
  _id: string;
  name: string;
  imageUrl: string;
}

export const Villager = ({ name, imageUrl, _id }: Props) => {
  console.log(_id)
  return (
    <div>
      <h1>{name}</h1>
      <img src={imageUrl} alt={name}/>
      <Link to={`/villagers/:${_id}`}>Learn More</Link>
    </div>
  )
}
