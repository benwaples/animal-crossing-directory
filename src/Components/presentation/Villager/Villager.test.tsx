import React from 'react'
import { render } from '@testing-library/react'
import { Villager } from './Villager'
import { MemoryRouter } from 'react-router-dom'

describe('Villager component snapshot', () => {
  it('should render a villager component', () => {
    const { asFragment } = render(
    <MemoryRouter>
      <Villager 
      name="agent 2"
      imageUrl="show me the villager"
      _id="villagerId"
      />
    </MemoryRouter>)

    expect(asFragment()).toMatchSnapshot()
  })
})