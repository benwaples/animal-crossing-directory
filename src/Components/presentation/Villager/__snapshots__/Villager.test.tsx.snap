// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component should render a villager component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      agent 2
    </h1>
    <img
      alt="agent 2"
      src="show me the villager"
    />
    <a
      href="/villagers/villagerId"
    >
      Learn More
    </a>
  </div>
</DocumentFragment>
`;
